src/sdl_font.cpp

```diff
@@ -90,91 +90,93 @@ std::unique_ptr<Font> Font::load_font( SDL_Renderer_Ptr &renderer, SDL_PixelForm
 void Font::draw_ascii_lines( const SDL_Renderer_Ptr &renderer, const GeometryRenderer_Ptr &geometry,
                              unsigned char line_id, const point &p, unsigned char color ) const
 {
+    int half_height = ( height / 2 ) - 1;
+    int half_width = ( width / 2 ) - 1;
     SDL_Color sdl_color = palette[color];
     switch( line_id ) {
         // box bottom/top side (horizontal line)
         case LINE_OXOX_C:
-            geometry->horizontal_line( renderer, p + point( 0, ( height / 2 ) ), p.x + width, 1,
+            geometry->horizontal_line( renderer, p + point( 0, half_height ), p.x + width, 1,
```

Comment:
For `height == 17`, this draws a horizontal line at `y == 7`, but it should probably draw the line at `y == 8` which is the true center of 17 pixels.

I think the real problem here is that the calculation of the x/y coordinates did not take into account of the line width, so it should subtract the line width before dividing by 2.

Reply:
oh yeah good point !